plugin/src/main/java/org/opensearch/sql/plugin/catalog/CatalogServiceImpl.java

```diff
@@ -117,16 +120,12 @@ private <T> T doPrivileged(PrivilegedExceptionAction<T> action) {
   }
 
   private StorageEngine createStorageEngine(CatalogMetadata catalog) throws URISyntaxException {
-    StorageEngine storageEngine;
     ConnectorType connector = catalog.getConnector();
     switch (connector) {
       case PROMETHEUS:
-        PrometheusClient
-            prometheusClient =
-            new PrometheusClientImpl(new OkHttpClient(),
-                new URI(catalog.getUri()));
-        storageEngine = new PrometheusStorageEngine(prometheusClient);
-        break;
+        return connectorTypeStorageEngineFactoryMap
+            .get(catalog.getConnector())
+            .getStorageEngine(catalog.getName(), catalog.getProperties());
```

Comment:
We may lazy eveluate the getStorageEngine when Catalog.getStorageEngine is called. The benfit is that the failure of storageEngine construction will impact requet, but  not impact Plugin itself.

Reply:
https://github.com/vamsi-amazon/sql/blob/catalog-enhancements/plugin/src/main/java/org/opensearch/sql/plugin/catalog/CatalogServiceImpl.java#L82

I am catching throwable here, this will make sure plugin construction won't get impacted.
Understood the concern that customer isn't properly communicated when catalog is misconfigured.

Will try to address the lazy evaluation in the coming release.